proposal/try-catch.md

```diff
@@ -0,0 +1,92 @@
+# Support `@try` and `@catch` flow controls for error handling: Draft 1
+
+*[(Issue)](https://github.com/sass/sass/issues/2619)*
+
+## Table of Contents
+
+## Background
+
+> This section is non-normative.
+
+In many situations, particularly when writing tests, it's useful to catch error
+and warning messages without breaking compilation. By adding `@try` and `@catch`
+at-rules similar to other control-flow syntax, authors will have more control
+over error handling, and the ability to test for error-cases.
+
+Many large Sass projects currently use wrapper functions and mixins to catch and
+manage errors, with a parameter to toggle between throwing or returning the
+error message. But that relies on the special error handling to be built into
+every individual function or mixin that might error. By adding global controls,
+error-handling can happen either where the error is generated, or where it is
+defined.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a pair of new at-rule directives that must be used
+together as a form of control. The `@try { ... } @catch { ... }` syntax captures
+any thrown output (error or warning messages) from inside the `@try` block, and
```

Comment:
I agree that catching warnings with this is unexpected. It's not how other programming languages work, and it may do very unexpected things with the control flow.

I'd like to punt on the question of how warnings are tested for now. It's much easier and safer to devise custom warning-capturing infrastructure, since you can just add strings to a list and you don't have to worry about globally coordinating return values.